### PR TITLE
cb-32264: fix signal handling for 371

### PR DESCRIPTION
**Pull request by zacharyestep**

### CODE REVIEW

File: cmd/cb-event-forwarder/main.go [hunk 1]
```diff
@@ -387,14 +389,11 @@ func messageProcessingLoop(uri, queueName, consumerTag string) error {
...
-				wg.Wait()
-				os.Exit(1)
-			}
+		case signal := <-outputSignals:
+			log.Errorf("%s- exiting immediately.", signal)
+			c.Shutdown()
+			wg.Wait()
+			os.Exit(1)
 
 		case closeError := <-c.connectionErrors:
```
Comment (jcapolino) [inline cmd/cb-event-forwarder/main.go:398]: Some customers are reporting errors where we're getting connectionErrors and then exiting as well.  We should probably not exit here?

Comment (jcapolino) [reply cmd/cb-event-forwarder/main.go:398]: We know it's this call and not the error call based on the logs.

Comment (zacharyestep) [reply cmd/cb-event-forwarder/main.go:398]: Case's are exclusive in golang so this should avoid exiting during runtime errors and only exit when there is an signal to the daemon to shutdown like SIGTERM

**Comment by jcapolino** [comment]
I approve this PR as well.

**Pull request merged**